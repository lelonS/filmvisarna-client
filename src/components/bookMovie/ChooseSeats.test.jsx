import React, { useState as useStateMock } from 'react'
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ChooseSeats from './ChooseSeats';

const exampleScreening = {
  "_id": "6554d8ffe1e758c3b57d55c5", "date": "2023-12-08", "time": "15:00", "theater": 1, "movieID": "6554d6d2e1e758c3b57d55c3", "theaterName": "Stora Salongen", "seats": [
    [{ "seat": false, "seatNumber": 1, "rating": 3 },
    { "seat": true, "seatNumber": 2, "rating": 4 },
    { "seat": false, "seatNumber": 3, "rating": 4 },
    { "seat": false, "seatNumber": 4, "rating": 4 }]]
}


jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}))



// seat: true = booked
// { booked: false, row: 1, seat: 1, seatNumber: 1 }

describe('ChooseSeats', () => {

  const setState = jest.fn();

  beforeEach(() => {
    useStateMock.mockImplementation(init => [init, setState]);
  });

  test('renders correctly when seat selected', () => {
    render(<ChooseSeats screening={exampleScreening} seats={[{ booked: false, row: 1, seat: 1, seatNumber: 1 }]} setSeats={() => { }} />);

    // Assert that the form elements are rendered correctly
    expect(screen.getByText(/Antal biljetter:/i)).toBeInTheDocument();
    expect(screen.getByText(/Rad:/i)).toBeInTheDocument();
    expect(screen.getByText(/Plats:/i)).toBeInTheDocument();
  });


  test('renders correctly when seat not selected', () => {
    // Override the useStateMock from the beforeEach (set the toggle to true which means that the seats are not available)
    useStateMock.mockImplementationOnce(() => [true, setState]);
    render(<ChooseSeats screening={{ ...exampleScreening }} seats={[]} setSeats={() => { }} />);

    // Assert that the form elements are rendered correctly
    expect(screen.getByText(/Antal biljetter:/i)).toBeInTheDocument();
    expect(screen.getByText(/Plats:/i)).toBeInTheDocument();
    expect(screen.getByText(/Välj andra säten, ett eller flera säten av dom du försökte välja är redan bokade/i)).toBeInTheDocument();
  });

  test('renders correctly when seat not selected snapshopt test', () => {
    // Override the useStateMock from the beforeEach (set the toggle to true which means that the seats are not available)
    useStateMock.mockImplementationOnce(() => [true, setState]);
    const tree = renderer.create(<ChooseSeats screening={{ ...exampleScreening }} seats={[]} setSeats={() => { }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly snapshopt test', () => {
    const tree = renderer.create(<ChooseSeats screening={{ ...exampleScreening }} seats={[{ booked: false, row: 1, seat: 1, seatNumber: 1 }]} setSeats={() => { }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});