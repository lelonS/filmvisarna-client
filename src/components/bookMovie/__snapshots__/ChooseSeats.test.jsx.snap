// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseSeats renders correctly snapshopt test 1`] = `
<div
  className="lg:w-[700px] md:w-[70%] w-[80%] container mt-5"
>
  <label
    className="flex items-center cursor-pointer"
    htmlFor="toogleA"
  >
    <div
      className="relative"
    >
      <input
        className="sr-only"
        id="toogleA"
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="w-10 h-4 bg-gray-400 rounded-full shadow-inner"
      />
      <div
        className="dot absolute w-6 h-6 bg-white rounded-full shadow -left-1 -top-1 transition"
      />
    </div>
    <div
      className="ml-3 text-white font-medium"
    >
      Välj separata platser
    </div>
  </label>
  <div
    className="screen mb-7 mt-4"
  />
  <div>
    <div
      className="flex justify-center w-full sm:gap-2 mb-2"
      id="row-1"
    >
      <button
        className="bg-footerGrey cursor-pointer seat lg:w-10 lg:h-7 md:w-8 md:h-8 w-5 h-5 "
        disabled={false}
        id="row1seat-1"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <button
        className="bg-red-600 seat lg:w-10 lg:h-7 md:w-8 md:h-8 w-5 h-5 "
        disabled={true}
        id="row1seat-2"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <button
        className="bg-footerGrey cursor-pointer seat lg:w-10 lg:h-7 md:w-8 md:h-8 w-5 h-5 "
        disabled={false}
        id="row1seat-3"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <button
        className="bg-footerGrey cursor-pointer seat lg:w-10 lg:h-7 md:w-8 md:h-8 w-5 h-5 "
        disabled={false}
        id="row1seat-4"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    </div>
  </div>
  <div
    className="text-white text-center mt-2 mb-4"
  >
    <p>
      Antal biljetter:  
    </p>
    <p>
      Rad: 
      1 -
       Plats:
       
      1
    </p>
  </div>
</div>
`;

exports[`ChooseSeats renders correctly when seat not selected snapshopt test 1`] = `
<div
  className="lg:w-[700px] md:w-[70%] w-[80%] container mt-5"
>
  <label
    className="flex items-center cursor-pointer"
    htmlFor="toogleA"
  >
    <div
      className="relative"
    >
      <input
        className="sr-only"
        id="toogleA"
        onChange={[Function]}
        type="checkbox"
      />
      <div
        className="w-10 h-4 bg-gray-400 rounded-full shadow-inner"
      />
      <div
        className="dot absolute w-6 h-6 bg-white rounded-full shadow -left-1 -top-1 transition"
      />
    </div>
    <div
      className="ml-3 text-white font-medium"
    >
      Välj separata platser
    </div>
  </label>
  <div
    className="screen mb-7 mt-4"
  />
  <div>
    <div
      className="flex justify-center w-full sm:gap-2 mb-2"
      id="row-1"
    >
      <button
        className="bg-footerGrey cursor-pointer seat lg:w-10 lg:h-7 md:w-8 md:h-8 w-5 h-5 "
        disabled={false}
        id="row1seat-1"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <button
        className="bg-red-600 seat lg:w-10 lg:h-7 md:w-8 md:h-8 w-5 h-5 "
        disabled={true}
        id="row1seat-2"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <button
        className="bg-footerGrey cursor-pointer seat lg:w-10 lg:h-7 md:w-8 md:h-8 w-5 h-5 "
        disabled={false}
        id="row1seat-3"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <button
        className="bg-footerGrey cursor-pointer seat lg:w-10 lg:h-7 md:w-8 md:h-8 w-5 h-5 "
        disabled={false}
        id="row1seat-4"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    </div>
  </div>
  <div
    className="text-white text-center mt-2 mb-4"
  >
    <p>
      Antal biljetter:  
    </p>
    <p>
      Plats:
       
    </p>
    <p
      className="text-red-500"
    >
      Välj andra säten, ett eller flera säten av dom du försökte välja är redan bokade
    </p>
  </div>
</div>
`;
